timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -89,11 +104,40 @@ def __init__(self, data: Any, *args, **kwargs):
                 "Data input type not recognized, must be DataFrame or iterable."
             )
         super().__init__(data=data, *args, **kwargs)
+        self._static_features: Optional[pd.DataFrame] = None
+        if static_features is not None:
+            self.static_features = static_features
 
     @property
     def _constructor(self) -> Type[TimeSeriesDataFrame]:
         return TimeSeriesDataFrame
 
+    @property
+    def _item_index(self) -> pd.Index:
+        return self.index.get_level_values(0).unique()
+
+    @property
+    def static_features(self):
+        return self._static_features
+
+    @static_features.setter
+    def static_features(self, value: Optional[pd.DataFrame]):
+        # if the current item index is not a multiindex, then we are dealing with a single
+        # item slice. this should only happen when the user explicitly requests only a
+        # single item or during `slice_by_timestep`. In this case we do not set static features
+        if not isinstance(self.index, pd.MultiIndex):
+            return
```

Comment:
Maybe `return None` here to be more explicit?

Reply:
I would think this is semantically a bit cleaner. See https://stackoverflow.com/questions/15300550/return-return-none-and-no-return-at-all 
So to my understanding is that we `return None` when we have an `Optional` return type and we're explicitly returning a `None` somebody else may do something with. And `return` just means stop evaluating the function. What do you think?

Reply:
Somehow my read of that StackOverflow answer was that `return None` would be more appropriate here, but this is a really minor nitpick from my side, so I guess either option is fine.